app/services/password_changer.go

```diff
@@ -24,5 +26,16 @@ func PasswordChanger(store data.AccountStore, r ops.ErrorReporter, cfg *app.Conf
 		return FieldErrors{{"credentials", ErrFailed}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
```

Comment:
So `POST /password` does two things. It trips me up every time.

1) `PasswordChanger` is called if the user has a session already. It's intended for use in an account settings page.
2) `PasswordResetter` is called if the user has a token instead of a session. It's intended for lost password flows.

Given that, I'm unsure about requiring TOTP here, after the user has logged in. I don't really see other sites doing it. Thoughts?

Reply:
Using github as an example, if you ask for a password reset and your account has 2fa enabled, when you click on the reset link you are asked for the 2fa code before you can enter a new password. If this behavior is handled by `PasswordResetter` then the 2fa should remain on that.

If you're already logged into Github you can change your password with your current password and a new password. There's no 2fa because you've presumably already demonstrated you have the secret when you got your session logging in. If this behavior is handled by `PasswordChanger` then I think the 2fa should be removed from this endpoint. (I currently have it on there)

I think it should be removed from `PasswordChanger`, want me to do that?

Reply:
> if you ask for a password reset and your account has 2fa enabled, when you click on the reset link you are asked for the 2fa code before you can enter a new password

okay, i'm happy to have precedent like that to follow.

> I think it should be removed from PasswordChanger, want me to do that?

👍